### PR TITLE
Update check-raid.rb

### PR DESCRIPTION
**Pull request by denogio**
## Pull Request Checklist

**Is this in reference to an existing issue?**

#### General

- [ ] Update Changelog following the conventions laid out on [Keep A Changelog](http://keepachangelog.com/)

- [ ] Update README with any necessary configuration snippets

- [ ] Binstubs are created if needed

- [ ] RuboCop passes

- [ ] Existing tests pass 

#### New Plugins

- [ ] Tests

- [ ] Add the plugin to the README

- [ ] Does it have a complete header as outlined [here](http://sensu-plugins.io/docs/developer_guidelines.html#coding-style)

### CODE REVIEW

**Comment by denogio** [comment]
Checking a MegaRaid device will create MegaSAS.log in the execution path "/opt/sensu".
Found out when the log filled my root file system. Adding -NoLog fixes the issue

**Comment by majormoses** [comment]
@dhpowrhost so this basically prevents there from being a logfile created. I have a few questions about this in general:
- would users want the log? even a small subset might want this. I think a user flag (even if we default to true) makes sense. 
- is this available on all versions or is this a somewhat newer feature?

**Comment by dajeffers** [comment]
I could see users wanting the log data, however live data is more important than historic data for identifying a hard drive's status. NoLog has been a feature in megacli for a while.

**Comment by majormoses** [comment]
ok, then can we add the no log option behind a cli arg? Something like this:
```ruby
# new option
 option :log,
         short: '-l VALUE',
         long: '--log VALUE',
         description: 'Enables or disables logging for megacli',
         boolean: true,
         default: true
# check and issue with the appropriate args
contents = if config[:log]
             `/usr/sbin/megacli -AdpAllInfo -aALL`
           else
             `/usr/sbin/megacli -AdpAllInfo -aALL -NoLog`
           end
```

This should get us the best of both worlds backwards compatibility and the ability to disable logging.

**Comment by denogio** [comment]
Could be fine with an option to enable logs, but i cant see any reason why it should be enabled by default. Should only be for debuggin purposes.

**Comment by majormoses** [comment]
The reason is backwards compatibility, unless there is a very good reason we avoid breaking changes. Also this mirrors the vendor defaults which is to create a log and an option to avoid doing so.

**Comment by dajeffers** [comment]
For what it's worth, a very good reason would be to avoid filling up a disk. I've run into numerous servers with the log filling a disk, as did @dhpowrhost. This varies from how the vendor runs it because the way that this runs in Sensu is going to be on a _much_ more frequent basis than an admin signing on here and there to check up on the health of the array.

**Comment by majormoses** [comment]
Now that is a very good reason! I am 👍 on having a default of `false`. This becomes a breaking change, so it needs to be called out as such in the changelog. I actually like breaking changes when they bring a better user experience, it is just that we need to think of how it impacts existing users and ensure proper justification for changing what they know is happening. Given the impact which is very minor as you pointed out most people will be running it quite frequently and therefore fill up disk quicker. There could still be impacted users that maybe run it daily and ship those logs to somewhere and this may break them. I have weighed the potential good vs bad and feel as long as it is properly versioned (on the maintainers end) and called out in the changelog in a clear way it is a very reasonable tradeoff. To those who do not pin and consult the changelog without upgrading that is on you. Still these are the things we consider as maintainers as we evaluate and make decisions whether something should be a backwards compatible change. 
Please refer to: https://github.com/sensu-plugins/community/blob/master/HOW_WE_CHANGELOG.md to craft a proper changelog entry.

Example:
```
## [Unreleased]
### Breaking Change
- `check-raid.rb`: added option `--log` with a default of `false` to prevent it default creating a log which is frequently being written to and filling up the disk. This does the *opposite* of what the vendor defaults are but due to the nature of those running it through a monitoring solution like sensu the defaults do not make sense in this use case. If you are wanting those logs you can change this to `true` to keep existing behavior.
```

**Comment by dajeffers** [comment]
@dhpowrhost any movement on the changelog request from @majormoses ?

**Comment by denogio** [comment]
@dajeffers 
This is my first pull request. I don't even know where to write the changelog ?

**Comment by majormoses** [comment]
@dhpowrhost in the repo there is a file named `CHANGELOG.md` that you will modify, we have a document to help people understand our changelog guidelines here: https://github.com/sensu-plugins/community/blob/master/HOW_WE_CHANGELOG.md. In the repo `CHANGELOG.md` file you will find a section that looks like this:
```
## [Unreleased]
```

Immediately below that insert these lines:
```
### Breaking Change
- `check-raid.rb`: added option `--log` with a default of `false` to prevent it default creating a log which is frequently being written to and filling up the disk. This does the *opposite* of what the vendor defaults are but due to the nature of those running it through a monitoring solution like sensu the defaults do not make sense in this use case. If you are wanting those logs you can change this to `true` to keep existing behavior.
```
The final result should look like this:
```
## [Unreleased]
### Breaking Change
- `check-raid.rb`: added option `--log` with a default of `false` to prevent it default creating a log which is frequently being written to and filling up the disk. This does the *opposite* of what the vendor defaults are but due to the nature of those running it through a monitoring solution like sensu the defaults do not make sense in this use case. If you are wanting those logs you can change this to `true` to keep existing behavior.
```
That is all you have to do, the rest will be handled by a maintainer after it merges. It will typically be released within 24 hours of approval.

Basically what it boils down to explaining the breaking impact to a consumer of the plugin (that means you) so that it is clear what they need to change to keep the existing behavior if that is needed.

**Comment by mbbroberg** [comment]
> This is my first pull request. I don't even know where to write the changelog ?

That's so cool. Thank you for contributing @dhpowrhost and congrats on the first. I hope @majormoses's last note helps clarify it. 

If you're still stuck, I'm happy to screenshare and work on it with you week or next week. Message me here or on [Sensu Community Slack](http://slack.sensu.io).

**Comment by majormoses** [comment]
closing in favor of #16

**Comment by majormoses** [comment]
released: https://rubygems.org/gems/sensu-plugins-raid-checks/versions/2.0.0
